developer/utils/caching.py

```diff
@@ -90,6 +92,34 @@ def cache_apps(
             logger.info("> found %s" % filename)
             env_filenames.append(os.path.join(env_path, filename))
 
+    processed = set()
+    for desc, bundle_type in _iterate_environment(env_filenames):
+        # This will avoid us from attempting to process
+        # the same descriptor twice.
+        if ("%s" % desc) in processed:
```

Comment:
is this the same as saying `repr(desc)`?

Reply:
yup. your version is better